index.js

```diff
@@ -4,6 +4,11 @@ module.exports = {
   },
 
   responseObject: function(response) {
+    try {
+      JSON.parse(response.xhr.responseText);
+    } catch(err) {
+      return {invalid_JSON: response.xhr.responseText};
```

Comment:
Ok. I think I see the actual problem is that templategallery isn't using this in a promise.